system/lib/webgpu/webgpu.cpp

```diff
@@ -27,3 +38,24 @@ WGPUInstance wgpuCreateInstance(const WGPUInstanceDescriptor* descriptor) {
 
 void wgpuInstanceReference(WGPUInstance) { /* no-op for now */ }
 void wgpuInstanceRelease(WGPUInstance) { /* no-op for now */ }
+
+// WGPUSurface
+
+void wgpuSurfaceGetCapabilities(WGPUSurface surface,
+                                WGPUAdapter adapter,
+                                WGPUSurfaceCapabilities* capabilities) {
+  WGPUTextureFormat preferredFormat =
+      wgpuSurfaceGetPreferredFormat(surface, adapter);
+  assert(preferredFormat == WGPUTextureFormat_BGRA8Unorm ||
+         preferredFormat == WGPUTextureFormat_RGBA8Unorm);
+  capabilities->formatCount = 3;
```

Comment:
nit: move this inside the `if` branches and use `.size()` so that the formatCount will be correct by construction.